src/Kernel/Context.class.st

```diff
@@ -2411,6 +2419,13 @@ Context >> unwindTo: aContext [
 		ctx := ctx findNextUnwindContextUpTo: aContext]
 ]
 
+{ #category : 'private' }
+Context >> willFailReturn [
```

Comment:
The thing is you **can** step at this point but you **don't want to** because the step would initiate #cannotReturn etc which is what the VM would do in this place - to make the step and fail to return etc. What we want here is to "peek ahead" and avoid this situation; it's an analogy of InstructionStream>>willReturn etc. However, this whole #stepToCalleeOrNil method is just a quick fix to replace #stepToCalee in #runUntilErrorOrReturnFrom: while keeping its previous behavior. IMO #stepToCalee should never have been used in #runUntilErrorOrReturnFrom: - #stepToCallee should simulate a step to callee or sender but was used as a cleanup instrument to get rid of those extra guard contexts instead. Much cleaner way would be IMO to use #releaseTo: or something like this but at this point I'd prefer this simple fix to avoid confusion :) (i.e. in the future, if this PR shows no problems, I'd remove this #stepToCalleeOrNil again and replace it with something more conceptual)